src/lib/rerender.js

```diff
@@ -19,20 +22,48 @@ module.exports = function(app) {
 function rerender(done) {
   const page = this.get('page');
 
-  // Prevent rerender if currently processing
-  if (page != null && page.containsState(RENDERED)) {
-    const { req, res } = this.getCurrentContext();
+  if (page == null) {
+    return;
+  }
+
+  // Queue rerender if currently processing
+  if (!page.containsState(RENDERED)) {
+    pending = page;
+    return;
+  }
+
+  const { req, res } = this.getCurrentContext();
+
+  renderPage(page, req, res, done);
+}
 
-    page.debug('rerendering');
-    page.appendState(-RENDERED, RENDERING);
+/**
+ * Render 'page'
+ * @param {Page} page
+ * @param {Request} req
+ * @param {Response} res
+ * @param {Function} done
+ */
+function renderPage(page, req, res, done) {
+  page.debug('rerendering');
+  page.appendState(-RENDERED, RENDERING);
+  clock.frame(() => {
```

Comment:
https://github.com/developit/preact/blob/576dda310a8e31e524fa2ecc803d6d406597c41e/src/util.js#L13 reverts to `setTimeout()` if `Promise` is not supported. On browsers that don't support `Promise` a `clock.frame()` might trigger too fast?

Reply:
I guess that would only be an issue for browsers that support `requestAnimationFrame` but not `Promise`, which is exactly 0.

Reply:
good point. We polyfill both, but they would also fall back to `setTimeout(0)`. Not sure what would happen then